compaction/compaction_manager.cc

```diff
@@ -1573,6 +1573,7 @@ future<compaction_manager::compaction_stats_opt> compaction_manager::perform_sst
             case sstables::compaction_type_options::scrub::quarantine_mode::only:
                 return sst->is_quarantined();
             }
+            on_internal_error(cmlog, "bad scrub quarantine mode");
```

Comment:
we are also using `__builtin_unreachable()` in this case. see https://github.com/scylladb/scylladb/blob/34d41177fe246731e446ee1ebb5856040e051299/types/types.cc#L910-L913 for an example. probably we can use it instead for better consistency? another advantage of using `__builtin_unreachable()` is that, once we can use C++23, we will be able to s/__builtin_unreachable()/std::unreachable()/`, see https://en.cppreference.com/w/cpp/utility/unreachable .

Reply:
It's safer to throw in non-critical paths to avoid undefined behavior.